### ISSUE TITLE
[BUG] Unable to use lexical-core in stable no_std environment

### ISSUE THREAD

**Issue by tstellanova** [Alexhuszagh/rust-lexical#37]
## Description

Include lexical-core as a dependency in Cargo.toml, and turn off std:

`lexical-core =  {version="0.7.4", default-features=false }`

Build gives an error:

```
error[E0554]: `#![feature]` may not be used on the stable release channel
   --> /Users/todd/.cargo/registry/src/github.com-1ecc6299db9ec823/lexical-core-0.7.4/src/lib.rs:133:35
    |
133 | #![cfg_attr(not(feature = "std"), feature(core_intrinsics))]
```

**Comment by myrrlyn** [comment]
This is not currently possible due to the dependency on floating-point intrinsics that are not present in `core::f64`.

I am not familiar enough with floating-point operations, either within this crate or within the language primitives, to feel comfortable altering the implementation here or copying implementations from libcore.

I believe that the current rationale for keeping numeric methods out of `core` is that their succinct operation requires the presence of a platform `libm` or implementations in `compiler-rt`? I expect that the `core::intrinsics` functions we are using forward to one of those two implementations but I haven't looked.

This is an unfortunate limitation and I will leave the ticket open, but that is my current understanding of our blocker.

**Comment by Alexhuszagh** [comment]
@myrrlyn It requires the LLVM intrinsics, specifically, in the following macros, which are then called with a lot of functions:

https://github.com/Alexhuszagh/rust-lexical/blob/18076323d1225c910c086f0b9fa56ac83458d40f/lexical-core/src/util/num.rs#L843

https://github.com/Alexhuszagh/rust-lexical/blob/18076323d1225c910c086f0b9fa56ac83458d40f/lexical-core/src/util/num.rs#L857

https://github.com/Alexhuszagh/rust-lexical/blob/18076323d1225c910c086f0b9fa56ac83458d40f/lexical-core/src/util/num.rs#L874

These methods are then invoked required the following intrinsics:

- fabsf32
- ceilf32
- expf32
- floorf32
- logf32
- powif32
- powf32
- roundf32
- fabsf64
- ceilf64
- expf64
- floorf64
- logf64
- powif64
- powf64
- roundf64

At least 2 of these intrinsics are actually required, so if you have alternate suggestions for `no_std`, please let me know. Also, sorry for the massive delay in dealing with issues, I've been on a mental health break.

The required uses (not comprehensive) can be found below:
https://github.com/Alexhuszagh/rust-lexical/blob/18076323d1225c910c086f0b9fa56ac83458d40f/lexical-core/src/atof/algorithm/correct.rs#L63
https://github.com/Alexhuszagh/rust-lexical/blob/18076323d1225c910c086f0b9fa56ac83458d40f/lexical-core/src/ftoa/radix.rs#L27

Only one of these is required when the `radix` feature is not active, and we currently only need fixes for `float::pow` and `float::ln`, where `float` is `f32` or `f64`. @tstellanova If you can find an alternative or these intrinsics are stabilized, I should be able to no longer require nightly for `no_std`.

**Comment by Alexhuszagh** [comment]
I've added a feature called `libm` which should remove this dependency, allowing `no_std` to work on stable as long as the `libm` feature is enabled. I will update the documentation accordingly, and this should be in the next release.

https://github.com/Alexhuszagh/rust-lexical/commit/5c41e9b94448086f17f504110e64e61311089f97

**Comment by Alexhuszagh** [comment]
This has been added to the [documentation ](features)and added to the Travis CI [builds](https://github.com/Alexhuszagh/rust-lexical/blob/master/.travis.yml#L23-L24) to ensure it works.

**Comment by Alexhuszagh** [comment]
Closed as of [v0.7.6](https://github.com/Alexhuszagh/rust-lexical/releases/tag/v0.7.6). You may now use the `libm` feature to enable stable, `no_std` functionality. I'm hoping versions of these intrinsics will be added to stable some day, so I will avoid adding libm as a default feature.
